### ISSUE TITLE
New rule: warn on bare/naked returns

### ISSUE THREAD

**Issue by chavacava** [mgechev/revive#115]
In GO, when using named return values it is possible to use naked (aka bare) returns.
For example:
```go
func ReturnId() (id int, err error) {
   id = 10
   return id, err
}
```
can be written as

```go
func ReturnId() (id int, err error) {
   id = 10
   return
}
```

This _feature_ is somewhat controversial, because it can lead to hard-to-understand code, see for example

* Nate Finch's [Go Nitpicks](https://npf.io/2014/10/go-nitpicks/)
* [proposal: Go 2: remove bare return](https://github.com/golang/go/issues/21291)

Why do not add a rule that warns on bare returns?

**Comment by mgechev** [comment]
I like that! I've seen this pattern frequently but if folks do not agree, they can just not enable the rule.

**Comment by chavacava** [comment]
> but if folks do not agree, they can just not enable the rule.

exactly

I'll try to do a PR this week

**Comment by inliquid** [comment]
You must use bare return if you use named return params. This will help to distinguish them. Using named returns when return params already named - this is what actually needs to be warned about.

**Comment by chavacava** [comment]
As said in the issue description, the subject is controversial.
Some people (including me) do not like bare returns because they find that code is less understandable.
On the other side, I love named results because, I think, they help in understanding the contract of a function (`revive` has a rule that suggests to add names to confusing results). As [Effective Go says](https://golang.org/doc/effective_go.html#named-results) "they're documentation".

Anyway, we can add another rule to warn when "using named returns when return params already named"
